packages/web-react/package.json

```diff
@@ -18,6 +18,8 @@
   "types": "./index.d.ts",
   "dependencies": {
     "@floating-ui/react": "^0.26.5",
+    "@lmc-eu/spirit-web": "^1.10.0",
+    "@lmc-eu/spirit-icons": "^1.1.1",
```

Comment:
I think you are right. I have been thinking about the Icons and move them to `optionalDependencies`. But `spirit-web` seems to me that is the direct dependency that is needed always.

Reply:
@crishpeen Good point 👍🏻 

Just for the record, I was thinking the `optionalDependencies` property would describe it better:

```json
"optionalDependencies": {
  "@lmc-eu/spirit-web": "^1.10.0",
  "@lmc-eu/spirit-icons": "^1.1.1",
}
```

- https://docs.npmjs.com/cli/v10/configuring-npm/package-json#optionaldependencies
- https://classic.yarnpkg.com/lang/en/docs/dependency-types/

But, as noted in the npm docs, we should check if **any** design tokens and icons package are installed and throw an error if they are not. No, I don't know how to do that 🙂. And, more importantly, it does not do what we want it to do: unreachable optional dependencies just do not cause the installation to fail, that's all.

---

So, having come this far, I think optional `peerDependencies` is fine. But a check for tokens and icons from our side (instead of a broken build) would still be a nice DX improvement.

Reply:
I agree. It would be nice to have some kind of check with instructions on how to fix it.

Reply:
Well, the icons have already been checked for missing icons in the map or an empty context. So I consider this as good enough. For web-react, there are instructions that you must install the web package as well. If you don't you will not see the styles. I do not know about any tool that would help us know if some styling has been provided and it is compatible with the web-react package.